lib/Reader/readerir.cpp

```diff
@@ -1320,10 +1334,70 @@ GenIR::getType(CorInfoType CorType, CORINFO_CLASS_HANDLE ClassHandle,
   }
 }
 
+#define FEATURE_AVX_SUPPORT 1
+
+bool canUseAVX() 
+{
+#ifdef FEATURE_AVX_SUPPORT
+  return 1;
+#else
+  return false;
+#endif
+}
+
+#define XMM_REGSIZE_BYTES        16      // XMM register size in bytes
+#define YMM_REGSIZE_BYTES        32      // YMM register size in bytes
+
+#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
+#define _TARGET_XARCH_
+#endif
+
+// Whether SSE2 is available
+bool canUseSSE2()
+{
+#ifdef _TARGET_XARCH_
+  return 1;//opts.compCanUseSSE2;
+#else
+  return false;
+#endif
+}
+
+unsigned getSIMDVectorRegisterByteLength()
+{
+#if defined(_TARGET_XARCH_) && !defined(LEGACY_BACKEND)
+  if (canUseAVX())
+  {
+    return YMM_REGSIZE_BYTES;
+  }
+  else
+  {
+    assert(canUseSSE2());
+    return XMM_REGSIZE_BYTES;
+  }
+#else
+  assert(!"getSIMDVectorRegisterByteLength() unimplemented on target arch");
+  unreached();
+#endif
+}
+
 Type *
 GenIR::getClassType(CORINFO_CLASS_HANDLE ClassHandle, bool GetAggregateFields,
                     std::list<CORINFO_CLASS_HANDLE> *DeferredDetailAggregates) {
-
+  if (doSimdIntrinsicOpt() && JitContext->JitInfo->isInSIMDModule(ClassHandle)) {
+    std::string ClassName = appendClassNameAsString(ClassHandle, true, false, false);
+    if (ClassName.compare(0, 22, "System.Numerics.Vector") == 0 && ClassName.length() == 23) {
+      switch (ClassName[22]) {
+      case '2':
+        return Vector2Ty;
+      case '3':
+        return Vector3Ty;
+      case '4':
+        return Vector4Ty;
+      default:
+        assert(UNREACHED);
```

Comment:
I don't think that we want to assert here--instead, it's probably better to fall back to the usual code path for structs. Removing the assert should be enough.